manager/node.go

```diff
@@ -128,8 +128,9 @@ func (m *VolumeManager) DeleteNode(name string) error {
 	}
 	condition := types.GetNodeConditionFromStatus(node.Status, types.NodeConditionTypeReady)
 	// Only could delete node from longhorn if kubernetes node missing
-	if condition.Status == types.ConditionStatusTrue || condition.Reason != types.NodeConditionReasonKubernetesNodeDown ||
-		node.Spec.AllowScheduling || len(replicas) > 0 || len(engines) > 0 {
+	if condition.Status == types.ConditionStatusTrue || (condition.Reason != types.NodeConditionReasonKubernetesNodeDown &&
```

Comment:
ManagerPodMissing is different from it has been converted to non-worker. It can be just temporarily, e.g during upgrade or just started up.

Btw, we should rename `KubernetesNodeDown` to `KubernetesNodeGone`.

And previous the condition for `NodeGone` is only we cannot find it. But if we found it's nonworker, we should also consider it gone.

Reply:
How can we identify if the node is non-worker? 
I was curious to know where do we maintain the role of the node? Could you please point me to the right attribute?

Reply:
It's kubernetes node. If you see `k get node` you will see the Roles.

This is Kubernetes node in the Longhorn node controller: https://github.com/rancher/longhorn-manager/blob/master/controller/node_controller.go#L387

Reply:
You would need to check Kubernetes API to find out where is the role stored and use that value for deciding if we should declare the node is gone (in additional to the node is really gone)

Reply:
An easy way is looking at `kubectl get node -o yaml`.